src/java.base/share/classes/java/lang/System.java

```diff
@@ -2148,7 +2148,7 @@ public void setCause(Throwable t, Throwable cause) {
             }
 
             public ProtectionDomain protectionDomain(Class<?> c) {
```

Comment:
This accessor has become pointless since the functinoal removal of SM, but it would be out of scope for this PR.

Reply:
Thanks for looking at this.  I didn't want to change this with this change.

Reply:
There may need to be some follow-on cleanup, e.g. I'm wondering if Lookup.cachedProtectionDomain is needed now.

Reply:
One of the reasons I wanted to move this out of Hotspot as a native call is that it might make further work with ProtectionDomain easier to do all in Java, except there's still a bit of coupling in the JVM with the name of the class and that it's passed through defineClass (resolve_from_stream) and initialized in the mirror.  So I guess that's still a lot.

Reply:
Aside from JVMTI (CFLH for example), is there anything left in the VM that needs this? The last param to JVM_DefineClassWithSource has the location from the code source if available.

Reply:
The VM doesn't need this but it carries it around because it's a parameter to JVM_DefineClass and DefineClassWithSource (second to last parameter).  CFLH and CDS from what I can tell have it for the same purpose - ultimately to assign it into the mirror.
There are some remaining code in the compilers (ci).  Not sure if they are needed without studying it more.